.github/workflows/sync.yml

```diff
@@ -0,0 +1,52 @@
+name: Sync
+
+on:
+  push:
+    branches: [master, staging]
+  workflow_dispatch:
+
+jobs:
+  sync-master:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master'
```

Comment:
I think you could move this `if` to just the deploy step just to reduce repetition. The rest of the steps are the same between the two.